nari/ext/act/parser/ability.py

```diff
@@ -9,6 +9,47 @@
 from nari.types.event import Event
 from nari.ext.act.exceptions import ActLineParsingException
 
+try:
+    from nari_act_rust import ability_from_params
+except ImportError:
+    def ability_from_params(timestamp: Timestamp, params: list[str]) -> Ability:
+        """Internal function of ability_from_logline"""
+        source_actor = Actor(*params[0:2])
+        ability = AbilityEvent(*params[2:4])
+        target_actor = Actor(*params[4:6])
+        action_effects = []
+        for i in range(0, 16, 2):
+            index = i + 6
+            action_effects.append(
+                action_effect_from_logline(params[index:index + 2])
+            )
+        actor_update_resources_and_positions(source_actor, params[22:32])
+        actor_update_resources_and_positions(target_actor, params[32:42])
+
+        sequence_id = int(params[42], 16)
+
+        return Ability(
+            timestamp=timestamp,
+            action_effects=action_effects,
+            source_actor=source_actor,
+            target_actor=target_actor,
+            ability=ability,
+            sequence_id=sequence_id,
+        )
+
+def actor_update_resources_and_positions(actor: Actor, params: list[str]):
+    """Updates the actor position and resources values based on ACT packet"""
+    # apparently when the target actor is 'none', then the *source* actor's resources will be empty will also be empty
+    # also apparently, other time(s) it will be blank just because /shrug
+    try:
+        actor.resources.update(
+            *[int(x) for x in params[0:6]]
+        )
+        actor.position.update(
+            *[float(x) for x in params[6:10]]
+        )
+    except ValueError:
+        pass
```

Comment:
Why tho? It makes no sense to even have the actor class, I'm not sure why we'd wanna roll 2 ACT events into a single implementation when there's no reason to use the data in the first place.

Reply:
It was more I did the same thing in rust when it came to the functions, so it wasn't as much scrolling in the python file when reading it.

Reply:
Updated the doc string to be more clear where this is used

Reply:
Prefer to just not do this, in both python and rust. This was never something the parser should be doing in the first place.

Reply:
Seeing how it's a set structure for the specific packets, I see no reason to have a worse time reading more lines with longer indents for the exact same functionality as separating it to a function.

Reply:
Catching 2 potential exceptions in a single block masks what failed since you're passing the exception. That's reason enough to separate.

Reply:
In this case, it's fine - this is what I did before when I was parsing, and there are two reason for `pass`-ing here:

1. There's no real exception to re-raise here: the values are empty in these cases
2. Since there's no data, there's no extra corrective action to take - hence the `pass`